### ISSUE TITLE
esp-idf 4.3 i2c clock choice is invalid, please check flag and frequency

### ISSUE THREAD

**Issue by Ekstrapolator** [mkfrey/u8g2-hal-esp-idf#1]
Hi,

I'm using esp-idf version 4.3, hardware is LILYGO® TTGO T-Camera ESP32 WROVER & OLED I was able to build example project but when I'm uploading it to the chip I'm getting error:

`E (20) i2c: i2c_param_config(645): i2c clock choice is invalid, please check flag and frequency`
`E (30) err: esp_err_t = 258`
`assertion "0 && "i2c_param_config(I2C_MASTER_NUM, &conf)"" failed: file "../main/u8g2_esp32_hal.c", line 144, function: u8g2_esp32_i2c_byte_cb`

I have used arduino example to check display and hardware, and it is working fine, please find code below and terminal output attached. 
[putty.log](https://github.com/mkfrey/u8g2-hal-esp-idf/files/6987172/putty.log)
```
#include <driver/gpio.h>


#include <driver/spi_master.h>
#include <esp_log.h>
#include <freertos/FreeRTOS.h>
#include <freertos/task.h>
#include <stdio.h>
#include <string.h>
#include <u8g2.h>

#include "sdkconfig.h"
#include "u8g2_esp32_hal.h"

// SDA - GPIO21
#define PIN_SDA 21

// SCL - GPIO22
#define PIN_SCL 22

static const char* TAG = "ssd1306";

void task_test_SSD1306i2c(void* ignore) {
  u8g2_esp32_hal_t u8g2_esp32_hal = U8G2_ESP32_HAL_DEFAULT;
  u8g2_esp32_hal.sda = PIN_SDA;
  u8g2_esp32_hal.scl = PIN_SCL;
  u8g2_esp32_hal_init(u8g2_esp32_hal);

  u8g2_t u8g2;  // a structure which will contain all the data for one display
  u8g2_Setup_ssd1306_i2c_128x32_univision_f(
      &u8g2, U8G2_R0,
      // u8x8_byte_sw_i2c,
      u8g2_esp32_i2c_byte_cb,
      u8g2_esp32_gpio_and_delay_cb);  // init u8g2 structure
  u8x8_SetI2CAddress(&u8g2.u8x8, 0x78);

  ESP_LOGI(TAG, "u8g2_InitDisplay");
  u8g2_InitDisplay(&u8g2);  // send init sequence to the display, display is in
                            // sleep mode after this,

  ESP_LOGI(TAG, "u8g2_SetPowerSave");
  u8g2_SetPowerSave(&u8g2, 0);  // wake up display
  ESP_LOGI(TAG, "u8g2_ClearBuffer");
  u8g2_ClearBuffer(&u8g2);
  ESP_LOGI(TAG, "u8g2_DrawBox");
  u8g2_DrawBox(&u8g2, 0, 26, 80, 6);
  u8g2_DrawFrame(&u8g2, 0, 26, 100, 6);

  ESP_LOGI(TAG, "u8g2_SetFont");
  u8g2_SetFont(&u8g2, u8g2_font_ncenB14_tr);
  ESP_LOGI(TAG, "u8g2_DrawStr");
  u8g2_DrawStr(&u8g2, 2, 17, "Hi nkolban!");
  ESP_LOGI(TAG, "u8g2_SendBuffer");
  u8g2_SendBuffer(&u8g2);

  ESP_LOGI(TAG, "All done!");

  vTaskDelete(NULL);
}

void app_main(void)
{
    xTaskCreate(&task_test_SSD1306i2c, "test", 4048, NULL, 2, NULL);
}
```

**Comment by Ekstrapolator** [comment]
Added line **memset(&conf, 0, sizeof(i2c_config_t));** after creation of structure conf, inside u8g2_esp32_hal.c 

```
      i2c_config_t conf;
      memset(&conf, 0, sizeof(i2c_config_t));
      conf.mode = I2C_MODE_MASTER;
```

**Comment by mkfrey** [comment]
Thank you for reporting this bug and finding a solution!

The `clk_flags` member of the `i2c_config_t` struct was not initialized, leading to an undefined value. I've now added a null initialization to the struct declaration.

So this should be fixed with commit 95107d8faf7726d50fee2c87b3838d40d9093754.
